web3.go

```diff
@@ -320,3 +358,82 @@ func toBlockNumArg(number *big.Int) string {
 	}
 	return hexutil.EncodeBig(number)
 }
+
+func DeployContract(ctx context.Context, client Client, privateKeyHex string, contractData string) (*Transaction, error) {
+	if len(privateKeyHex) > 2 && privateKeyHex[:2] == "0x" {
+		privateKeyHex = privateKeyHex[2:]
+	}
+	privateKey, err := crypto.HexToECDSA(privateKeyHex)
+	if err != nil {
+		return nil, fmt.Errorf("invalid private key: %v", err)
+	}
+
+	gasPrice, err := client.GetGasPrice(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get gas price: %v", err)
+	}
+
+	publicKey := privateKey.Public()
+	publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+	if !ok {
+		return nil, errors.New("error casting public key to ECDSA")
+	}
+
+	fromAddress := crypto.PubkeyToAddress(*publicKeyECDSA)
+	nonce, err := client.GetPendingTransactionCount(ctx, fromAddress)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get nonce: %v", err)
+	}
+	decodedContractData, err := hexutil.Decode(contractData)
+	if err != nil {
+		return nil, fmt.Errorf("cannot decode contract data: %v", err)
+	}
+	//TODO try to use native Transaction only; can't sign currently
```

Comment:
I think that by extending `types.Signer` to work on interfaces, then we'll be able to use our native type only.